src/test/scala/de/bripkens/ha/ConfigurationSpec.scala

```diff
@@ -0,0 +1,72 @@
+package de.bripkens.ha
+
+import java.nio.file.{NoSuchFileException, Paths}
+
+import com.fasterxml.jackson.databind.JsonMappingException
+import org.scalatest.{EitherValues, Matchers, WordSpec}
+
+class ConfigurationSpec extends WordSpec with Matchers with EitherValues {
+
+  "Configuration" when {
+
+    "loading a valid config file" should {
+
+      val path = Paths.get(getClass.getResource("test-config.yml").toURI)
+      val loaded = Configuration.load(path)
+      loaded should be('right)
+      val configuration = loaded.right.get
+
+      "have two endpoints" in {
+        configuration.endpoints should have size 2
+      }
+
+      "have a shopping endpoint" in {
+        val expected = HealthCheckEndpoint("http://127.0.0.1:8081/healthcheck", "shopping", "Shopping System", 3000, "mySlackReporter")
+        configuration.endpoints should contain(expected)
+      }
+
+      "have a recommendation endpoint" in {
+        val expected = HealthCheckEndpoint("http://127.0.0.1:8181/healthcheck", "recommendation", "Recommendation System", 10000, "myConsoleReporter")
+        configuration.endpoints should contain(expected)
+      }
+
+      "have two reporters" in {
+        configuration.reporters should have size 2
+      }
+
+      "have a slack reporter" in {
+        val expected = SlackReporterConfig(null, "test", "https://hooks.slack.com/services/12345", "Health Check", "http://lorempixel.com/64/64/")
```

Comment:
Interesting. Seems like the Jackson sub type identification works based on the JSON property name. On the other hand, the Scala module does not seem to respect the `JsonProperty` annotation. Would be interesting to see what happens when the property name is changed from `reporterType` to just `type` in the configuration classes.